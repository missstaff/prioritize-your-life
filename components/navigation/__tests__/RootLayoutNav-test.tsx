import React from "react";
import { render, waitFor } from "@testing-library/react-native";
import RootLayout from "@/app/_layout";

jest.mock("expo-splash-screen", () => ({
  preventAutoHideAsync: jest.fn(),
  hideAsync: jest.fn(),
}));

jest.mock("expo-font", () => ({
  useFonts: jest.fn(),
}));

jest.mock("@expo/vector-icons", () => ({
  FontAwesome: {
    font: {},
  },
}));

jest.mock("@/components/navigation/RootLayoutNav", () => {
  return jest.fn(() => <div>Mocked RootLayoutNav</div>);
});

describe("RootLayout", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders RootLayoutNav when fonts are loaded", async () => {
    const { useFonts } = require("expo-font");
    const { preventAutoHideAsync, hideAsync } = require("expo-splash-screen");

    useFonts.mockReturnValue([true, null]);

    const { getByText } = render(<RootLayout />);
    expect(preventAutoHideAsync).toHaveBeenCalled();

    await waitFor(() => {
      expect(hideAsync).toHaveBeenCalled();
    });

    // expect(getByText("Mocked RootLayoutNav")).toBeTruthy(); //???
  });

  it("returns null when fonts are not loaded", () => {
    const { useFonts } = require("expo-font");

    useFonts.mockReturnValue([false, null]);

    const { toJSON } = render(<RootLayout />);

    expect(toJSON()).toBeNull();
  });

  it("throws an error if font loading fails", () => {
    const { useFonts } = require("expo-font");

    useFonts.mockReturnValue([false, new Error("Font loading error")]);

    expect(() => render(<RootLayout />)).toThrow("Font loading error");
  });

  it("matches snapshot when fonts are loaded", async () => {
    const { useFonts } = require("expo-font");
    const { hideAsync } = require("expo-splash-screen");

    useFonts.mockReturnValue([true, null]);

    const { toJSON } = render(<RootLayout />);

    await waitFor(() => {
      expect(hideAsync).toHaveBeenCalled();
    });

    expect(toJSON()).toMatchSnapshot();
  });
});
