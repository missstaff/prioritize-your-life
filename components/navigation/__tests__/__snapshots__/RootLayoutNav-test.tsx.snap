// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout matches snapshot when fonts are loaded 1`] = `
<div>
  Mocked RootLayoutNav
</div>
`;
